posthog/hogql/printer.py

```diff
@@ -1467,26 +1507,63 @@ def visit_window_expr(self, node: ast.WindowExpr):
                 raise ImpossibleASTError(f"Invalid frame method {node.frame_method}")
             if node.frame_start and node.frame_end is None:
                 strings.append(self.visit(node.frame_start))
-
             elif node.frame_start is not None and node.frame_end is not None:
                 strings.append("BETWEEN")
                 strings.append(self.visit(node.frame_start))
                 strings.append("AND")
                 strings.append(self.visit(node.frame_end))
-
             else:
                 raise ImpossibleASTError("Frame start and end must be specified together")
         return " ".join(strings)
 
     def visit_window_function(self, node: ast.WindowFunction):
         identifier = self._print_identifier(node.name)
-        exprs = ", ".join(self.visit(expr) for expr in node.exprs or [])
-        args = "(" + (", ".join(self.visit(arg) for arg in node.args or [])) + ")" if node.args else ""
+        exprs = [self.visit(expr) for expr in node.exprs or []]
+
+        # For compatibility with postgresql syntax, convert lag/lead to lagInFrame/leadInFrame and add default window frame if needed
+        if identifier in ("lag", "lead"):
```

Comment:
This should probably get a `if self.dialect == 'clickhouse':` somewhere, as when printing just HogQL (mostly done for debugging or little snippets), we can just print the function calls as they are.